### DISCUSSION TITLE
send text message

### DISCUSSION THREAD

**Discussion by symonjim** [emqx/MQTTX#1047] [Ideas]
Newbie. Sorry if this is already a feature I haven't found or a common and despised suggestion.

Can MQTTX send my phone a text message in response to getting an MQTT message? I've seen mqttwarn, MQTT Push Client and similar but I like MQTTX.

**Comment by ysfscream** [comment]
> Newbie. Sorry if this is already a feature I haven't found or a common and despised suggestion.
> 
> Can MQTTX send my phone a text message in response to getting an MQTT message? I've seen mqttwarn, MQTT Push Client and similar but I like MQTTX.

Thanks for the MQTT X like, it sounds like a great idea. But it's not currently supported.

Right now MQTT X is just a more pure MQTT client, the base functionality provided is connection publishing and subscriptions, it doesn't provide for subsequent application extensions after receiving a message, which I'm not so sure there's a lot of demand for that.

However, MQTT X is currently implementing a plugin system, and we would like to be able to extend the functionality to whatever the user wants to achieve through such a plugin, for example, for sending an SMS to a phone or posting a tweet.

**Reply by symonjim** [reply]
Thanks. Looking forward to a plugin system.
